internal/publisher/service.go

```diff
@@ -0,0 +1,73 @@
+package publisher
+
+import (
+	"fmt"
+
+	"github.com/emmaLP/gs-software-onboarding/internal/model"
+	"github.com/emmaLP/gs-software-onboarding/internal/queue"
+	"github.com/emmaLP/gs-software-onboarding/pkg/hackernews"
+	"go.uber.org/zap"
+)
+
+type service struct {
+	logger      *zap.Logger
+	hnClient    hackernews.Client
+	queueClient queue.Client
+}
+
+type Client interface {
+	processStories()
+}
+
+type ServiceOptions func(*service)
+
+func NewService(logger *zap.Logger, config *model.Configuration, queueClient queue.Client, opts ...ServiceOptions) (*service, error) {
+	hnClient, err := hackernews.New(config.Publisher.BaseUrl, nil)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create HackerNew client: %w", err)
+	}
+
+	service := &service{
+		logger:      logger,
+		hnClient:    hnClient,
+		queueClient: queueClient,
+	}
+
+	for _, opt := range opts {
+		opt(service)
+	}
+
+	return service, nil
+}
+
+func WithHackerNewsClient(client hackernews.Client) ServiceOptions {
```

Comment:
As the hackernews client is always required, this may make more sense to customise the `http.Client` rather than the entire `hackernews.Client`

Reply:
Although it's always requires, this option give me the ability to inject a mock hacknew client for testing